onnx/onnx-data.in.proto

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) ONNX Project Contributors.
+// Licensed under the MIT license.
+
+syntax = "proto2";
+
+package {PACKAGE_NAME};
+// #if ONNX-ML
+import "onnx/onnx-ml.proto";
+// #else
+import "onnx/onnx.proto";
+// #endif
+
+// This file contains the proto definitions for MapProto and
+// SequenceProto. These protos are used to represent the data structures
+// of maps and sequences in test data or can be used by ModelProto.
+
+// Sequences and maps are made of SequenceMapElement values, which can be
+// tensors, maps, or sequences.
+message SequenceMapElement {
+  enum DataType {
+    TENSOR = 1;
+    SPARSE_TENSOR = 2;
+    SEQUENCE = 3;
+    MAP = 4;
+  }
+  // The data type of the element.
+  // This field MUST have a valid SequenceMapElement.DataType value
+  optional int32 elem_type = 1;
+
+  optional TensorProto tensor_value = 2;
+  optional SparseTensorProto sparse_tensor_value = 3;
+  optional SequenceProto sequence_value = 4;
+  optional MapProto map_value = 5;
+}
+
+// Sequences
+//
+// Defines a dense, ordered, collection of elements that are of homogeneous types.
+message SequenceProto {
+
+  optional string name = 1;
+  repeated SequenceMapElement values = 2;
```

Comment:
In this way, there may be a tensor and a map in one sequence, which I don't think is correct.

Reply:
What if we do something similar to AttributeProto: define the DataType here, and use "repeated TensorProto", "repeated SparseTensorProto", etc.? Would that be better?

Reply:
I think Rama's suggested structure makes a lot of sense. In an offline discussion, Rama explained the strongly typed nature of these data structures in ONNX and how they operate differently than in Python (where you can have {'key':'value', 1: 32} as a valid dictionary), which was the reason behind this original design.

Reply:
As per my earlier interpretation, having a tensor and a map in one sequence was a valid option. I understand now that it should not be allowed. :)